src/Microsoft.AspNet.Mvc.Xml/XmlDataContractSerializerInputFormatter.cs

```diff
@@ -19,6 +20,7 @@ namespace Microsoft.AspNet.Mvc.Xml
     public class XmlDataContractSerializerInputFormatter : InputFormatter
     {
         private DataContractSerializerSettings _serializerSettings;
+        private ConcurrentDictionary<Type, object> _serializerCache = new ConcurrentDictionary<Type, object>();
```

Comment:
Static is a no go. If the user wants to go per request, he is responsible for caching. If we really cared we will have to provide another singleton service just for this. Because at this point he is already customizing it